research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+document, enabling the user to perform some task in a timely and immediate fashion. `popup` is
+similar to the `dialog` element, but is distinguished by its transient nature (“light dismiss”
+behaviors) and mutual exclusivity: only one `popup` element may be rendered at a time, with the
+exception of nested `popup`s.
+
+<p class="note">
+  Examples of popups include: button menus, listboxes used for item selection e.g. comboboxes,
+  "teaching" UI, and other progressive-disclosure widgets or content pickers.
+</p>
+
+## Showing a `popup` element
+
+A `popup` element should not be shown to the user unless:
+
+- The `initiallyopen` attribute is specified
+- The `show()` method is invoked
+- Or the `popup` element has been invoked via the `popup` attribute.
+
+This requirement may be implemented indirectly through the style layer. For example,
+user agents that
+[support the suggested default rendering](https://html.spec.whatwg.org/multipage/infrastructure.html#renderingUA)
+implement this requirement using the CSS rules described in the
+[Rendering section](https://html.spec.whatwg.org/multipage/rendering.html#rendering).
+
+### The `initiallyopen` attribute
+
+The `initiallyopen` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that the `popup` element should be shown on document load, such that
+the user can interact with it.
+
+**Example:**
+
+```html
+<popup initiallyopen></popup>
+```
+
+If the `initiallyopen` attribute is specified on a `popup` element _subject_, the user agent must
+run [showing a `popup` element steps](#showing-popup-steps).
+
+<p class="question">
+  When following these run steps, this effectively means the last popup in the DOM order with
+  initiallyopen specified will be the popup that "wins". Is that clear enough to implementers? Is
+  that a reasonable behavior?
+</p>
+
+### The `show()` method
+
+The `show()` method displays the `popup` element, such that the user can interact with it.
+
+**Example:**
+
+```html
+<popup id="newFeatureUI">
+  <p>
+    <strong>New!</strong>
+    I’m some sort of educational UI…
+  </p>
+</popup>
+
+<script>
+  if (upsellNewFeature) {
+    document.getElementById('newFeatureUI').show()
+  }
+</script>
+```
+
+When `show()` is invoked on a `popup` element _subject_, the user agent must
+run [showing a `popup` element steps](#showing-popup-steps).
+
+### The `popup` attribute
+
+The `popup` attribute indicates that a related `popup` element should be shown or hidden as the
+user interacts with the element where the `popup` attribute is specified. If the attribute is
+specified, the attribute's value must be the ID of a `popup` element in the same
+[node document](https://dom.spec.whatwg.org/#concept-node-document).
+
+**Example:** Invoking the `button` element in this example should show/hide the `popup` element,
```

Comment:
Hmm, that makes sense that light dismissal would be what actually hides the popup.

> Click twice, and the popup will still be showing, because it will first light-dismiss, and then the click will cause it to show again.

Could we come up with a way to prevent light dismissal in this case? e.g. don't light dismiss if the event target's attribute has an IDREF relationship with the popup. Otherwise, the `popup` attribute would lead to bad UX on any control that should toggle visibility of the popup, and IMHO is nearly useless.